From 6283ecae92ea209d50e2baf6c9b328f13e525e27 Mon Sep 17 00:00:00 2001
From: Roman Meshkevich <roman.meshkevich@cogentembedded.com>
Date: Wed, 14 Sep 2016 21:40:39 +0300
Subject: [PATCH 5/5] vsp-renderer: save returned length and bytesused from
 v4l2 driver when capture

If using buffer for output is used intermediate buffer, it is necessary
that one buffer bytesused and length is set non-zero.
---
 src/vsp-renderer.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/vsp-renderer.c b/src/vsp-renderer.c
index e985609..98abdc8 100644
--- a/src/vsp-renderer.c
+++ b/src/vsp-renderer.c
@@ -715,6 +715,16 @@ vsp_queue_buffer(int fd, int capture, struct vsp_surface_state *vs)
 		return -1;
 	}
 
+	if (capture) {
+		for (i = 0; i < vs->base.num_planes; i++) {
+			vs->base.planes[i].length = buf.m.planes[i].length;
+			/* XXX:
+			   Set length value to bytesused because bytesused is
+			   returned 0 from kernel driver. Need to set returned
+			   bytesused value. */
+			vs->base.planes[i].bytesused = buf.m.planes[i].length;
+		}
+	}
 	return 0;
 }
 
-- 
2.7.4

